tests/core/rename-property.test.ts

```diff
@@ -0,0 +1,83 @@
+import { renameObjectProperty } from '../../cocos/core/utils/internal';
+
+test(`renameObjectProperty()`, () => {
+    const symKey = Symbol();
+
+    const originalObject: Record<PropertyKey, any> = {
+        c: 1, a: '2', b: null,
+        get x() { return 6; },
+        [symKey]: false,
+    };
+    Object.defineProperty(originalObject, 'pNonEnumerable', {
+        value: 6,
+        enumerable: false,
+    });
+    Object.setPrototypeOf(originalObject, {
+        pNonOwn: '',
+    });
+    // Freeze to ensure our readonly on original object.
+    Object.freeze(originalObject);
+
+    const originalKeyOrder = Object.keys(originalObject);
+    expect(originalKeyOrder).toStrictEqual(['c', 'a', 'b', 'x']);
+
+    // Failed: original key does not exists.
+    expect(renameObjectProperty(originalObject, 'd', 'e')).toBe(originalObject);
+    
+    // Failed: new key has already existed.
+    expect(renameObjectProperty(originalObject, 'b', 'a')).toBe(originalObject);
+
+    // Failed: can not rename a non-enumerable object.
+    expect(renameObjectProperty(originalObject, 'pNonEnumerable', 'pNonEnumerable_1')).toBe(originalObject);
+
+    // Failed: can not rename a non-own object.
+    expect(renameObjectProperty(originalObject, 'pNonOwn', 'pNonOwn_1')).toBe(originalObject);
+
+    /// The following should all rename successively.
+
+    const checkOk = (originalPropertyKey: PropertyKey, newPropertyKey: PropertyKey) => {
+        const renamed = renameObjectProperty(originalObject, originalPropertyKey, newPropertyKey);
+
+        // A copy is returned instead of modifying original object.
+        expect(renamed).not.toBe(originalObject);
+
+        const expected: Record<PropertyKey, any> = {
+            ...originalObject,
+            [newPropertyKey]: originalObject[originalPropertyKey],
+        };
+        delete expected[originalPropertyKey];
```

Comment:
New property is added at the end of `expected`. How can it be the same order as original object?